src/IO/GEHDF5Wrapper.cxx

```diff
@@ -349,30 +362,35 @@ Succeeded GEHDF5Wrapper::initialise_geo_factors_data(const std::string& path,
 {
     if(path.size() == 0)
     {
-        if(is_signa)
+        if(is_signa) // AB todo "is_RDF9"
         {
             m_address = "/SegmentData/Segment4/3D_Norm_Correction/slice";
             if(slice_num > 0)
             {
+                // Open Dataset and get Dataspace(metadata)
                 std::ostringstream datasetname;
                 datasetname << m_address << slice_num;
                 m_dataset_sptr.reset(new H5::DataSet(file.openDataSet(datasetname.str())));
                 m_dataspace = m_dataset_sptr->getSpace();
                 // PW here I output the dataspace dimensions and order to be correctly translated in the main code.
-                int rank = m_dataspace.getSimpleExtentNdims();
-                hsize_t dims_out[2];
-          //      int ndims = m_dataspace.getSimpleExtentDims( dims_out, NULL);
-                     std::cout << "rank " << rank << ", dimensions " <<
-                         (unsigned long)(dims_out[0]) << " x " <<
-                         (unsigned long)(dims_out[1]) << std::endl;
-//                m_memspace_ptr = new H5::DataSpace()
-            }
-            //! \todo Get these numbers from the HDF5 file
-            {
-                m_NX_SUB = 1981;    // hyperslab dimensions
-                m_NY_SUB = 357;
-                m_NX = 1981;        // output buffer dimensions
-                m_NY = 357;
+                // AB todo get rid of PW comment?
+                // Read dimensions
+                const int rank = m_dataspace.getSimpleExtentNdims();
+                hsize_t dims[2];
+                m_dataspace.getSimpleExtentDims( dims, NULL);
+                // AB todo : remove all this? seems clutter. 
+            //      int ndims = m_dataspace.getSimpleExtentDims( dims_out, NULL);
+                        std::cout << "rank " << rank << ", dimensions " <<
+                            (unsigned long)(dims[0]) << " x " <<
+                            (unsigned long)(dims[1]) << std::endl;
+    //                m_memspace_ptr = new H5::DataSpace()
+
```

Comment:
MATLAB may have transposed the row and column so I would suggest to make sure if everything is read in correct order.